tests/Fixer/Phpdoc/PhpdocReturnSelfReferenceFixerTest.php

```diff
@@ -270,4 +271,54 @@ public function test(): self {
 ',
         ];
     }
+
+    public function testThisStartWithDollarMark(): void
+    {
+        $this->fixer->configure([
+            'replacements' => [
+                '$this' => 'self',
```

Comment:
`$this` has different semantic meaning than `self`
`$this` is exactly this instance
`self` is any instance of given class.

I believe the proposed change is not bringing improvement, as docs after change will be less accurate.